src/kirby/components/segmented-control/segment-item.ts

```diff
@@ -0,0 +1,9 @@
+export interface SegmentItem {
+  text: string;
+  id: string;
+  checked: boolean;
+  badge?: {
+    content: string;
+    type: 'primary' | 'secondary' | 'tertiary' | 'alert' | 'success' | 'danger';
```

Comment:
Consider to extract a common typescript type for these `type`s (I have a suspicion that they well be used over and over).